landing_page/spec/controllers/user_controller_spec.rb

```diff
@@ -0,0 +1,124 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+
+    context "create a new user account with acceptable credentials" do
+
+        it "successfully adds a new user" do
+            expect{create_new_user}.to change(User, :count).by(1)
+        end
+
+        it "logs the user in upon account creation" do
+            create_new_user
+            expect(session[:user_id]).to_not be_nil
+        end
+
+        it "redirects a new user to their profile page upon creation" do
+            create_new_user
+            expect(response).to redirect_to(assigns(:user))
+        end
+
+    end
+
+    context "FactoryBot tests" do
+
+    before :each do
+        @user = FactoryBot.create(:user)
+        session[:user_id] = @user.id
+        login(@user)
+    end
+
+        context "GET #index" do
+
+            it "assigns users" do
+                get :index
+                expect(assigns(:users)).to eq([@user])
+            end
+
+            it "renders the template for the index" do
+                get :index
+                expect(response).to render_template('index')
+            end
+
+            it "returns a succesful response" do
+                get :index
+                expect(response.status).to eq(200)
+            end
```

Comment:
This one could live separately as-is but I would suggest it being merged to the first scenario. My reasoning is, it has the same setup as the the other scenario and actually it's part of it, which means that we're sending the exact same request. 

If we merge them we would still be able to identify individually what failed based on the line number and that would not increase the test readability nor complexity, which would be valid points for having two separate scenarios.

Reply:
Resolved in commit 4871a5281315b569ef730a49585feaab4de96af4